packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -363,11 +364,12 @@ class IconButton extends StatelessWidget {
         hoverColor: hoverColor ?? theme.hoverColor,
         highlightColor: highlightColor ?? theme.highlightColor,
         splashColor: splashColor ?? theme.splashColor,
-        radius: splashRadius ?? math.max(
-          Material.defaultSplashRadius,
-          (effectiveIconSize + math.min(padding.horizontal, padding.vertical)) * 0.7,
-          // x 0.5 for diameter -> radius and + 40% overflow derived from other Material apps.
-        ),
+        radius: splashRadius ?? iconTheme.splashRadius ??
```

Comment:
Can you verify that setting this property will allow the BackButton and Menu button in the app bar to get the updated splash radius, since those are created automatically by the app bar widget.

Reply:
Thanks so much for the review! Appreciate it

When I apply `iconTheme` for `AppBar` only, hamburger manu, IconButton, and BackButton splash radius are updated

<details> 
<summary>minimal code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Material App',
      theme: ThemeData.dark().copyWith(
        appBarTheme: const AppBarTheme(
          iconTheme: IconThemeData(
            splashRadius: 22,
          ),
        ),
      ),
      home: const Home(),
    );
  }
}

class Home extends StatelessWidget {
  const Home({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      drawer: const Drawer(),
      endDrawer: const Drawer(),
      appBar: AppBar(
        title: const Text('Home App Bar'),
        actions: <Widget>[
          IconButton(
            onPressed: () {},
            icon: const Icon(Icons.more_vert),
          )
        ],
      ),
      body: Center(
        child: IconButton(
          icon: const Icon(Icons.next_plan),
          onPressed: () {
            Navigator.push(
                context, MaterialPageRoute(builder: (_) => Second()));
          },
        ),
      ),
    );
  }
}

class Second extends StatelessWidget {
  const Second({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Second App Bar'),
      ),
    );
  }
}

``` 
	
</details>

### Without  `iconTheme` for `AppBar`


https://user-images.githubusercontent.com/48603081/144258541-807c6202-63ec-40cc-93c5-ced0a931c00d.mov


### With  `iconTheme` for `AppBar`

https://user-images.githubusercontent.com/48603081/144258550-c36d9946-6ecb-4f39-90eb-d46c9597acd8.mov